### PR TITLE
copt deduplicate: embedded_css, embedded_fonts

### PR DESCRIPTION
**Pull request by hius07**
Main discussion in https://github.com/koreader/koreader/pull/10763.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/10876)

### CODE REVIEW

**Comment by hius07** [comment]
https://github.com/koreader/koreader/blob/9b40bb17b14220a0b6ff33f471c904d6db1f15f0/frontend/apps/reader/modules/readertypeset.lua#L276-L282
I kept the structure of the module, but these ^^ small methods are not used anywhere else and can be just inserted into their callers:
https://github.com/koreader/koreader/blob/9b40bb17b14220a0b6ff33f471c904d6db1f15f0/frontend/apps/reader/modules/readertypeset.lua#L106-L112

**Review by poire-z — APPROVED** [review]

**Comment by poire-z** [comment]
> but are called from onReadSettings differently

There may be no real answer or logic - depends on the people who implemented each over the years :/ and their preference between shortcuts or always using a single method :) (I'm not sure in which camp you are :)

> ie, is this UpdatePos really needed for smooth_scaling and nightmode_images.

Possibly not, but we will need a redraw. Usually, UpdatePos is dumb/clever enough to ask crengine to compute its hashes to see if a rerendering is needed, and tell frontend that - and UpdatePos should do the right thing in both cases.

**Comment by hius07** [comment]
> but we will need a redraw

I'm talking about `onReadSettings` only.
Maybe it's better to call `UpdatePos` once in the end, instead of calling it after reading each setting.

**Comment by poire-z** [comment]
> Maybe it's better to call UpdatePos once in the end, instead of calling it after reading each setting.

It shouldn't be needed in the onReadSettings() phase - and it is actually a no-op:
https://github.com/koreader/koreader/blob/f46f341b9b0e1a95884050680036ac5912d341a2/frontend/apps/reader/modules/readerrolling.lua#L1026-L1035

So, I guess that if you see it called, it's because the same individual methods used for setting settings after init are called while in onReadSettings - and it shouldn't hurt. So, it comes down to the 2 camps I mentionned above :)

File: frontend/apps/reader/modules/readertypeset.lua [hunk 1]
```diff
@@ -32,33 +32,14 @@ function ReaderTypeset:onReadSettings(config)
...
-        local global = G_reader_settings:readSetting("copt_embedded_fonts")
-        self.embedded_fonts = (global == nil or global == 1) and true or false
-    end
+    -- default to enable embedded fonts
     -- As this is new, call it only when embedded_fonts are explicitely disabled
-    -- self.ui.document:setEmbeddedFonts(self.embedded_fonts and 1 or 0)
-    if not self.embedded_fonts then
+    if self.configurable.embedded_fonts == 0 then
         self.ui.document:setEmbeddedFonts(0)
     end
```
Comment (poire-z) [inline frontend/apps/reader/modules/readertypeset.lua:39]: "At this is new..." is my writting from 2017 :) and I was probably very cautious at the time.
It feels now that you don't need that if, and you could just set it always, even if it's 1.

**Review by poire-z — APPROVED** [review]

**Comment by poire-z** [comment]
Should probably not block this work, but pinging @yparitcher the author (and may be only user ? :) of the docsettingtweak.koplugin plugin: can users be affected with these deduplications, if they use the removed version of the setting - and how likely they would be to use the removed or the kept setting names. Or are all the settings @hius07 is playing with not really candidates for user per-directory customisation ?

**Comment by yparitcher** [comment]
Currently i set both eitherway (for example: `font_size` & `copt_font_size`) one does the actual change and one gets it to show up in the bottom menu.

**Pull request merged**